src/io/ipc/write/writer.rs

```diff
@@ -153,12 +158,3 @@ impl<'a, W: Write> FileWriter<'a, W> {
         Ok(())
     }
 }
-
-/// Finish the file if it is not 'finished' when it goes out of scope
-impl<'a, W: Write> Drop for FileWriter<'a, W> {
```

Comment:
In general we need this or the file is not closed e.g. in the case of an error when writing or something, no?

Reply:
The initial reason I noticed this `Drop` is by a compilation error :

```
   |         self.writer.into_inner().map_err(|e| Error::from(e).into())
   |         ^^^^^^^^^^^
   |         |
   |         cannot move out of here
   |         move occurs because `self.writer` has type `std::io::BufWriter<W>`, which does not implement the `Copy` trait

```

Then I think we could just remove it since if the `FileWriter` is not finished intentionally as the user requires, we are OK to leave a corrupt written file.
1. for user program crash during writing, the file is not correct no matter we finish it or not.
2. for users who forget to finish before exiting, we should train them to add a `finish` intentionally. instead of finish it `secretly`. 😉